### DISCUSSION TITLE
Is it possible to use range of commits in cherry-pick?

### DISCUSSION THREAD

**Discussion by lolgear** [libgit2/libgit2#6034] [Q&A]
Cherry picking allows to use range of commits.
Does this feature exist in libgit2?

**Comment by ethomson** [comment]
Not yet!  This is one of those sort of interesting places where there are two ways to do the same thing in git.  git has a sequencer for cherry-picking commits (`rebase --merge`) and a sequencer for cherry-picking commits (`cherry-pick` with a range).  We currently only support rebase, but I can imagine that we would want to support cherry-picking multiple commits in a sequencer as well.  I think that our rebase implementation should guide us, and certainly there's some code reuse possible.  I don't know if we extract the sequencer into an abstract piece or if the two are just too conceptually different.
